components/brave_ads/core/internal/ads/serving/eligible_ads/exclusion_rules/notification_ads/notification_ad_exclusion_rules.cc

```diff
@@ -27,6 +28,9 @@ ExclusionRules::ExclusionRules(
 
   per_hour_exclusion_rule_ = std::make_unique<PerHourExclusionRule>(ad_events);
   exclusion_rules_.push_back(per_hour_exclusion_rule_.get());
+
+  embedding_exclusion_rule_ = std::make_unique<EmbeddingExclusionRule>();
+  exclusion_rules_.push_back(embedding_exclusion_rule_.get());
```

Comment:
This should be moved to the base class so that it is supported for all creative types

Reply:
I see your point. Although as things stand right now, v3 for the other serving types has not been implemented to use embeddings and it's actually just falling back to v2. So even in v3, embedding is not currently a requirement.

I think this also ties in into the decoupling discussion. We should probably not have a single, fixed version for all types of serving and the same for exclusion rules. Wdyt?

Reply:
Each ad type can now have its own serving version, this was a recent change in preparation for the modular ad serving.

Reply:
That's great. Should we remove the v3 for inline and new tab page then? Right now they are copy and paste from v2

Reply:
That would be great! :-)